main/acle.md

```diff
@@ -1888,6 +1889,146 @@ or compiler-generated functions.
+runtime makes the selection at load time.
+
+The following attributes trigger the multi version code generation:
+`__attribute__((target_version("name")))` and `__attribute__((target_clones("name",...)))`.
+
+1. These attributes can't be mixed with each other.
+2. The `"default"` version means the version of the function that would be generated
+without these attributes.
+3. `"name"` is the dependent features from the tables below.
+4. The dependent features could be joined by the `+` sign.
+5. None of these attributes will enable the corresponding ACLE feature(s) associated
+to the `"name"` expressed in the attribute.
+6. These attributes have no effect on the calling convention, all versions must
+use the same calling convention.
+7. If only the `"default"` version exist it should be linked directly.
+8. FMV may be disabled in compile time. In this case the `"default"`
+version shall be used.
+
+The attribute `__attribute__((target_version("name")))` expresses the following:
+
+1. when applied to a function it becomes one of the versions. Function with the same
+name may exist with multiple versions in the same translation unit.
+2. One `"default"` version of the function is required to be provided.
+   a. Implicitly, without this attribute,
+   b. or explicitly providing the `"default"` in the attribute.
+4. All instances of the versions shall share the same function signature.
+
+The attribute `__attribute__((target_clones("name",...)))` expresses the following:
+
+1. when applied to a function the compiler emits multiple versions based on the
+arguments.
+   a. One of them is implicitly the `"default"`.
+   b. If the `"default"` matches with an other explicitly provided version the
+compiler can emit only one function instead of the two.
+2. If a name is not recognized the compiler should ignore it.
+
+`__ARM_FEATURE_FUNCTION_MULTI_VERSIONING` is defined to 1 if the versioning
+mechanism described in this section is available in the compiler and it is enabled.
+
+### Name mangling
+
+The `"default"` version is not mangled top of the languge specific name mangling.
+
+Function version to be mangled for C/C++. `<vendor-specific suffix>` should hold the version
+information [[cxxabi]](#cxxabi) as `_(M<name>)*` `name` is from tables below.
+
+If multiple features are requested then those shall be appended in priority order.
+
+For example:
+``` c
+__attribute__((target_clones("crc32", "aes+sha1")))
+int foo(){..}
+```
+will produce these mangled names for C languae: `foo`, `foo._Mcrc32`, `foo._Msha1Maes`.
+
+
+### Mapping
+
+The following table lists the architectures feature mapping for AArch32
+
+   | **Priority** | **Architecture name**            | **Name**        | **Dependent feature registers** |
+   | ------------ | -------------------------------- | --------------- | ------------------------------- |
+   | 0            | N/A                              | default         | ```N/A```                       |
```

Comment:
I think that the triple ` is not needed here in this table... I think it would be best to get to single `